src/NUnitFramework/tests/Constraints/PredicateConstraintTests.cs

```diff
@@ -52,5 +55,46 @@ public void CanUseConstraintExpressionSyntax()
         {
             Assert.That(123, Is.TypeOf<int>().And.Matches<int>((int x) => x > 100));
         }
+
+        [TestCase(typeof(object))]
+        [TestCase(typeof(string))]
+        [TestCase(typeof(int?))]
+        [TestCase(typeof(AttributeTargets?))]
+        public static void ActualMayBeNullForNullableTypes(Type type)
+        {
+            // https://github.com/nunit/nunit/issues/1215
+            var methodInfo = typeof(PredicateConstraintTests)
+                .GetMethods(BindingFlags.NonPublic | BindingFlags.Static)
+                .Single(method => method.Name == nameof(ActualMayBeNullForNullableTypes) && method.GetParameters().Length == 0)
+                .MakeGenericMethod(type);
+
+            ((Action)methodInfo.CreateDelegate(typeof(Action))).Invoke();
+        }
+
+        private static void ActualMayBeNullForNullableTypes<T>()
+        {
+            Assert.That(null, new ConstraintExpression().Matches<T>(actual => true));
+        }
+
+        [TestCase(typeof(int))]
+        [TestCase(typeof(AttributeTargets))]
+        public static void ActualMustNotBeNullForNonNullableTypes(Type type)
+        {
+            // https://github.com/nunit/nunit/issues/1215
+            var methodInfo = typeof(PredicateConstraintTests)
+                .GetMethods(BindingFlags.NonPublic | BindingFlags.Static)
+                .Single(method => method.Name == nameof(ActualMustNotBeNullForNonNullableTypes) && method.GetParameters().Length == 0)
```

Comment:
Isn't the test for the number of parameters redundant since there is only one matching method?